test/Worker.Extensions.Tests/ServiceBus/ServiceBusSessionMessageConverterTests.cs

```diff
@@ -0,0 +1,121 @@
+            {
+                BindingData = input;
+            }
+
+            public override IReadOnlyDictionary<string, object?> BindingData { get; }
+        }
+
+        internal sealed class TestFunctionContext : FunctionContext
+        {
+            public TestFunctionContext(BindingContext bindingContext)
+            {
+                BindingContext = bindingContext;
+            }
+
+            public override BindingContext BindingContext { get; }
+
+            public override string InvocationId => throw new NotImplementedException();
+
+            public override string FunctionId => throw new NotImplementedException();
+
+            public override TraceContext TraceContext => throw new NotImplementedException();
+
+            public override RetryContext RetryContext => throw new NotImplementedException();
+
+            public override IServiceProvider InstanceServices { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+            public override FunctionDefinition FunctionDefinition => throw new NotImplementedException();
+
+            public override IDictionary<object, object> Items { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+            public override IInvocationFeatures Features => throw new NotImplementedException();
+        }
+
+        [Fact]
+        public async Task ConvertAsync_ReturnsSuccess()
+        {
+            var data = "{\"SessionLockedUntil\":\"2024-12-05T21:10:36.1193094+00:00\"}";
+            
+            var bindingDataDictionary = new Dictionary<string, object>
+            {
+                { "SessionId", "test" },
+                { "SessionActions", JsonSerializer.Serialize(new
+                    {
+                        SessionLockedUntil = "2024-12-05T21:10:36.1193094+00:00"
+                    })
+                }
+            };
+            var context = new TestConverterContext(typeof(ServiceBusSessionMessageActions), data, new TestFunctionContext(new TestBindingContext(bindingDataDictionary)));
+            var converter = new ServiceBusSessionMessageActionsConverter(new MockSettlementClient("test"));
+            var result = await converter.ConvertAsync(context);
+
+            Assert.Equal(ConversionStatus.Succeeded, result.Status);
+            var output = result.Value as ServiceBusSessionMessageActions;
+            Assert.NotNull(output);
+        }
+
+        [Fact]
+        public async Task ConvertAsync_Batch_ForReceivedMessage_ReturnsSuccess()
+        {
+            var data = "{\"SessionLockedUntil\":\"2024-12-05T21:10:36.1193094+00:00\"}";
+
+            IList<string> repeatedField = new List<string> { "test" };
+
+            var bindingDataDictionary = new Dictionary<string, object>
+            {
+                { "SessionIdArray", repeatedField },
+                { "SessionActions", JsonSerializer.Serialize(new
```

Comment:
I don't actually know how these sessions work in SB, but for a batch scenario like this, can we have multiple actions? Id like for us to test batch properly with more than just `SessionIdArray` if that's a thing?

Reply:
There are two action types: `ServiceBusMessageActions` and `ServiceBusSessionMessageActions`.

In a batch scenario the former is an array - one for each incoming message. In the later it is only a single element. Session + Batch will be for a bath of messages _all from the same session_ - so a single session actions works. It is an implementation oddity that the binding data is an array of session IDs, as it will just be an array of the same repeated session ID.

Reply:
Yeah we can only test with `SessionIdArray` for the batched scenario since that's how the binding data is formatted today